ldfparser/parser.py

```diff
@@ -104,128 +156,226 @@ def parse_real_or_integer(self, i:str):
-
-	def ldf_node_atributes(self,s):
-		return
-		# return {"ldf_node_atributes" : "NOT_IMPLEMENTED"}
-	def ldf_node_atributes_node(self,s):
-		return
-		# return {"ldf_node_atributes_node" : "NOT_IMPLEMENTED"}
+	def ldf_identifier(self, tree):
+		return tree[0][0:]
 
-	def ldf_schedule_table(self,s):
-		return
-		# return {"ldf_schedule_table" : "NOT_IMPLEMENTED"}
-	def ldf_signal_representation(self,s):
-		return {"representations" : s}
-	def ldf_signal_representation_node(self,s):
-		return s
+	def ldf_integer(self, tree):
+		return self.parse_integer(tree[0])
+
+	def ldf_float(self, tree):
+		return self.parse_real_or_integer(tree[0])
+
+	def start(self, tree):
+		return tree[0]
+
+	def ldf(self, tree):
+		ldf = {}
+		for k in tree[0:]:
+			ldf[k[0]] = k[1]
+		return ldf
+
+	def header_lin_description_file(self, tree):
+		return ("header", "lin_description_file")
+
+	def header_protocol_version(self, tree):
+		return ("protocol_version", tree[0])
+
+	def header_language_version(self, tree):
+		return ("language_version", tree[0])
+
+	def header_speed(self, tree):
+		return ("speed", int(float(tree[0]) * 1000))
+
+	def header_channel(self, tree):
+		return ("channel_name", tree[0])
+
+	def nodes(self, tree):
+		return ("nodes", {'master': tree[0], 'slaves': tree[1]})
+
+	def nodes_master(self, tree):
+		return {"name": tree[0], "timebase": tree[1] * 0.001, "jitter": tree[2] * 0.001}
+
+	def nodes_slaves(self, tree):
+		return tree
+
+	def node_compositions(self, tree):
+		return ("node_compositions", tree[0:])
+
+	def node_compositions_configuration(self, tree):
+		return {"name": tree[0], "compositions": tree[1]}
+
+	def node_compositions_composite(self, tree):
+		return {"name": tree[0], "nodes": tree[1:]}
+
+	def signals(self, tree):
+		return ("signals", tree)
+
+	def signal_definition(self, tree):
+		return {"name": tree[0], "width": int(tree[1]), "init_value": tree[2], "publisher": tree[3], "subscribers": tree[4:]}
+
+	def signal_default_value(self, tree):
+		return tree[0]
+
+	def signal_default_value_single(self, tree):
+		return int(tree[0])
+
+	def signal_default_value_array(self, tree):
+		return tree[0]
+
+	def frames(self, tree):
+		return ("frames", tree)
+
+	def frame_definition(self, tree):
+		return {"name": tree[0], "frame_id": int(tree[1]), "publisher": tree[2], "length": tree[3] if len(tree) > 4 else None, "signals": tree[4] if len(tree) > 4 else tree[3]}
+
+	def frame_signals(self, tree):
+		return tree[0:]
+
+	def frame_signal(self, tree):
+		return {"signal": tree[0], "offset": int(tree[1])}
+
+	def sporadic_frames(self, tree):
+		return ("sporadic_frames", tree[0:])
+
+	def sporadic_frame_definition(self, tree):
+		return {"name": tree[0], "frames": tree[1:]}
+
+	def event_triggered_frames(self, tree):
+		return ("event_triggered_frames", tree[0:])
+
+	def event_triggered_frame_definition(self, tree):
+		return {"name": tree[0]}
+
+	def event_triggered_frame_definition_frames(self, tree):
+		return tree[0:]
+
```

Comment:
Missing diagnostic frame definitions, at worst it should transform it into an empty array, like:

```python
def diagnostic_frames(self, tree):
		return ("diagnostic_frames", [])
```